pkg/test/resourcefixture/testdata/basic/bigqueryreservation/v1alpha1/bigqueryreservationreservation/failovertononfailoverreservation/_http.log

```diff
@@ -0,0 +1,171 @@
+Content-Type: application/json
+User-Agent: kcc/${kccVersion} (+https://github.com/GoogleCloudPlatform/k8s-config-connector) kcc/controller-manager/${kccVersion}
+X-Goog-Request-Params: name=projects%2F${projectId}%2Flocations%2Fus-west2%2Freservations%2Fbigqueryreservationreservation${uniqueId}
+
+404 Not Found
+Content-Type: application/json; charset=UTF-8
+Server: ESF
+Vary: Origin
+Vary: X-Origin
+Vary: Referer
+X-Frame-Options: SAMEORIGIN
+X-Xss-Protection: 0
+
+{
+  "error": {
+    "code": 404,
+    "message": "No reservation found.",
+    "status": "NOT_FOUND"
+  }
+}
+
+---
+
+POST https://bigqueryreservation.googleapis.com/v1/projects/${projectId}/locations/us-west2/reservations?%24alt=json%3Benum-encoding%3Dint&reservationId=bigqueryreservationreservation${uniqueId}
+Content-Type: application/json
+User-Agent: kcc/${kccVersion} (+https://github.com/GoogleCloudPlatform/k8s-config-connector) kcc/controller-manager/${kccVersion}
+X-Goog-Request-Params: parent=projects%2F${projectId}%2Flocations%2Fus-west2
+
+{
+  "autoscale": {
+    "maxSlots": "50"
+  },
+  "edition": 3,
+  "secondaryLocation": "us-east1"
+}
+
+200 OK
+Content-Type: application/json; charset=UTF-8
+Server: ESF
+Vary: Origin
+Vary: X-Origin
+Vary: Referer
+X-Frame-Options: SAMEORIGIN
+X-Xss-Protection: 0
+
+{
+  "autoscale": {
+    "maxSlots": "50"
+  },
+  "creationTime": "123456789",
+  "edition": 3,
+  "name": "projects/${projectId}/locations/us-west2/reservations/bigqueryreservationreservation${uniqueId}",
+  "originalPrimaryLocation": "us-west2",
+  "primaryLocation": "us-west2",
+  "secondaryLocation": "us-east1",
+  "updateTime": "2024-04-01T12:34:56.123456Z"
+}
+
+---
+
+GET https://bigqueryreservation.googleapis.com/v1/projects/${projectId}/locations/us-west2/reservations/bigqueryreservationreservation${uniqueId}?%24alt=json%3Benum-encoding%3Dint
+Content-Type: application/json
+User-Agent: kcc/${kccVersion} (+https://github.com/GoogleCloudPlatform/k8s-config-connector) kcc/controller-manager/${kccVersion}
+X-Goog-Request-Params: name=projects%2F${projectId}%2Flocations%2Fus-west2%2Freservations%2Fbigqueryreservationreservation${uniqueId}
+
+200 OK
+Content-Type: application/json; charset=UTF-8
+Server: ESF
+Vary: Origin
+Vary: X-Origin
+Vary: Referer
+X-Frame-Options: SAMEORIGIN
+X-Xss-Protection: 0
+
+{
+  "autoscale": {
+    "maxSlots": "50"
+  },
+  "creationTime": "123456789",
+  "edition": 3,
+  "name": "projects/${projectId}/locations/us-west2/reservations/bigqueryreservationreservation${uniqueId}",
+  "originalPrimaryLocation": "us-west2",
+  "primaryLocation": "us-west2",
+  "secondaryLocation": "us-east1",
+  "updateTime": "2024-04-01T12:34:56.123456Z"
+}
+
+---
+
+PATCH https://bigqueryreservation.googleapis.com/v1/projects/${projectId}/locations/us-west2/reservations/bigqueryreservationreservation${uniqueId}?%24alt=json%3Benum-encoding%3Dint&updateMask=slotCapacity%2CignoreIdleSlots%2Cconcurrency%2Cautoscale
```

Comment:
Something seems off with this PATCH request. The `updateMask` includes fields that haven’t changed in `update.yaml`, while missing some that have (such as `secondaryLocation`). Could we double-check the controller logic for updates?